pandas/core/groupby.py

```diff
@@ -38,6 +38,23 @@
 import pandas.algos as _algos
 import pandas.hashtable as _hash
 
+
+class _ErrorContainer(Exception):
+    """Used internally to encapsulate an error. If need be, it should be caught
```

Comment:
pls take this out. this should not be part of this issue and is much too complicated for an already complicated operation.  What is the test case you actually need this for?